### PR TITLE
Docs: document Block and BlockCall

### PR DESCRIPTION
**Pull request by certik**

### CODE REVIEW

File: doc/src/asr/asr_nodes/statement_nodes/blockcall.md [hunk 1]
```diff
@@ -12,17 +12,20 @@ BlockCall(int label, symbol m)
...
 
 None.
 
 ## Description
 
-**BlockCall** signifies the block symbol call. Find more information on
-[block](../symbol_nodes/symbol.md).
+**BlockCall** signifies the block symbol call. The Block represents a
+scoped collection of statements and its own symbol table. The Block itself is
+part of a symbol table, and the BlockCall node is used to call it.
```
Comment (czgdp1807) [inline doc/src/asr/asr_nodes/statement_nodes/blockcall.md:26]:
```suggestion
part of a symbol table, and the BlockCall node is used to call it in order to execute the statements inside the block i.e., once the control reaches the `BlockCall` only then the statements inside the block it refers to are executed.
```

**Review by czgdp1807 — COMMENTED** [review]
Just some explanation on how `BlockCall` node helps in timing the execution of statements inside the block.

**Comment by Shaikh-Ubaid** [comment]
Could we also add few statements about how a Block and BlockCall are different from Function and FunctionCall?

**Comment by certik** [comment]
> Could we also add few statements about how a Block and BlockCall are different from Function and FunctionCall?

Definitely! Can you please submit a PR once this is merged?

I am trying to answer all questions that anybody has about ASR as PRs in this documentation.

**Comment by Shaikh-Ubaid** [comment]
>Can you please submit a PR once this is merged?

Done https://github.com/lfortran/lfortran/pull/1572.

**Pull request merged**